### PR TITLE
FullCalendar

### PR DESCRIPTION
**Pull request by achillesecos**
Parent: https://github.com/googleinterns/step78-2020/commit/249e9fecfc00bfe1195605055ebb5d1d1fe38c3b

The changes here will display the hard coded schedules from the prototype on FullCalendar. Will need to change that once input form is merged, so that it fetches the data from the input forms.

### CODE REVIEW

File: src/main/java/com/google/sps/Scheduler.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/Scheduler.java]: Don't forget to adjust this to be `getLectureSection` instead of `getlectureSection`

File: frontend/src/script.js
Comment (hjohns28) [inline frontend/src/script.js]: It doesn't seem like you're using all of these methods from the old google calendar display for full calendar - might want to delete the unnecessary ones if so

File: src/main/java/com/google/sps/server/FullCalendarDisplay.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/server/FullCalendarDisplay.java:46]: As a general thing, this logic is handled in the frontend branch already. No need to make changes here, but once we merge the json returne/fetch request will be to `/handleUserInput` (`ScheduleCreationHandler.java`) servlet. Again, no need to change this for now tho.

**Review by hjohns28 — APPROVED** [review]
LGTM pending comment (plus there are some console.log() statements you could remove)

File: src/main/java/com/google/sps/server/ScheduleCalendar.java
Comment (maceonthompson) [inline src/main/java/com/google/sps/server/ScheduleCalendar.java]: See comment in `Scheduler.java`, `getlectureSection` &rightarrow; `getLectureSection` (at least for the final pr into master)

File: src/test/java/com/google/sps/RankTest.java
Comment (maceonthompson) [inline src/test/java/com/google/sps/RankTest.java:647]: After merging, delete unused commented out code

**Review by maceonthompson — APPROVED** [review]
LGTM pending the getLectureSection change/a few nits!

File: frontend/src/script.js
Comment (achillesecos) [reply frontend/src/script.js]: This is weird, I had removed these functions... Maybe its on a different commit?

File: src/main/java/com/google/sps/Scheduler.java
Comment (achillesecos) [reply src/main/java/com/google/sps/Scheduler.java]: Thanks for the catch. Pretty sure its updated in a different commit, I'll check it out.

File: src/main/java/com/google/sps/server/FullCalendarDisplay.java
Comment (achillesecos) [reply src/main/java/com/google/sps/server/FullCalendarDisplay.java:46]: Yea that sounds good. Once we merge things together, we could redirect the fetch to the `/handleUserInput`

File: src/main/java/com/google/sps/server/ScheduleCalendar.java
Comment (achillesecos) [reply src/main/java/com/google/sps/server/ScheduleCalendar.java]: Thanks again. I'll find the correct commit that has these changes

File: src/test/java/com/google/sps/RankTest.java
Comment (achillesecos) [reply src/test/java/com/google/sps/RankTest.java:647]: Sounds good.

File: frontend/src/script.js
Comment (kristen-law) [inline frontend/src/script.js]: Should this be `'/displayFullCalendar'`?

File: frontend/src/Calendar.js
Comment (kristen-law) [inline frontend/src/Calendar.js]: The variable and function names here are inconsistent - I'm guessing one of them should be changed to Sunday/Monday?

File: frontend/src/Calendar.js
Comment (kristen-law) [inline frontend/src/Calendar.js:11]: No need to change anything, but you can avoid the need to add this as a separate constant/comment by documenting the unit in the variable name (e.g. startTimeMs). You'll also see `meetingTime.start * 60 * 1000` sometimes to more explicitly show the calculation.

**Review by kristen-law — CHANGES_REQUESTED** [review]

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Thanks for pointing this out. I changed it to Sunday, since this is technically the start of the week, rather than Monday.

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js:11]: Thanks for pointing this out.

File: frontend/src/App.js
Comment (FabriceCastel) [inline frontend/src/App.js]: Is there a reason for this div? It doesn't seem to do anything here.

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: nit: `selectedScheduleId` or `selectedScheduleIndex` or similar

-> id or index over idx
-> being more specific about what this schedule index really means

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: Change not necessary, but JS gives you a more concise way to express this kind of for-loop:

```
schedules.forEach(this.getMeetingTimes);
```

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: These methods aren't getters, they'd be more aptly named `selectNext/PreviousSchedule`

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: If you have a function `foo()`, then

```
() => { foo(); }
() => foo()
foo
```

Are all equivalent. You can replace these with, eg. `onClick={this.getNextSchedule}` and it will avoid instantiating an extra unnecessary lambda.

Additionally, please do not use spaces between properties and their values when declaring HTML elements, ie. `onClick={...}` and not `onClick = {...}`

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: Instead of tracking whether the fetch completed in the state, you could check to see if the current schedule index is `< schedulesTimes.length`. It would mean you have less information to keep track of and your render method won't break if you happen to fetch zero calendars from the backend.

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: This function name doesn't reflect what the function does

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: `getScheduleSections` -> `generateCourseMeetingTimes` or something more indicative of what this does

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: -> `selectNextSchedule()`

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: -> `selectPreviousSchedule()`

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: `schedulesTimes:  this.state.schedulesTimes.concat(combinedMeetingTimes)`

File: frontend/src/Calendar.js
Comment (FabriceCastel) [inline frontend/src/Calendar.js]: nit: indentation

**Review by FabriceCastel — CHANGES_REQUESTED** [review]

File: frontend/src/script.js
Comment (achillesecos) [reply frontend/src/script.js]: Done.

File: frontend/src/App.js
Comment (achillesecos) [reply frontend/src/App.js]: True, I used to have buttons in there, but forgot to remove the div when I got rid of them.

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Thanks for the suggestion. I went with `selectedScheduleId`

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Done

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Done

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Changed name to `generateScheduleMeetingTimes() `

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Done. Changed name.

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: I believe I tried using the `concat` method, but it did not work.

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Thanks. Changed to those names.

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: I see what you mean. Originally I tried that but was running into [this](https://www.andrewconnell.com/blog/deal-with-undefined-this-react-event-handler-performant-way/) issue. I will also fix the `onClick=` spacing error

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Got it. Thank you

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: This is a better approach. I changed it to `this.state.selectedScheduleId < this.state.schedulesTimes.length`

File: frontend/src/Calendar.js
Comment (achillesecos) [reply frontend/src/Calendar.js]: Fixed.

File: frontend/src/Calendar.js
Comment (FabriceCastel) [reply frontend/src/Calendar.js]: Is `combinedMeetingTimes` an array? If yes, then that would explain it. If not, then `concat()` should return a new array containing everything in the first array, plus the element you gave it as a param

Comment (FabriceCastel) [reply frontend/src/Calendar.js]: The reason this matters is that `concat` behaves differently if the param you give it is an array.

```
const oneTwo = [1, 2];
const threeFour = [3, 4];
const three = 3;

oneTwo.concat(threeFour) => [1, 2, 3, 4]
[ ...oneTwo, threeFour ] => [1, 2, [3, 4]]
[ ...oneTwo, ...threeFour ] = [1, 2, 3, 4]

oneTwo.concat(three) => [1, 2, 3]
[ ...oneTwo, three ] = [1, 2, 3]
```

Comment (achillesecos) [reply frontend/src/Calendar.js]: Yes it is an array, so that explains it

File: frontend/src/Calendar.js
Comment (FabriceCastel) [reply frontend/src/Calendar.js]: While this solution works, you should consider binding these functions in your component's constructor instead. If you do, then what I've suggested will work just fine and will save you the overhead of instantiating new lambdas.

Comment (achillesecos) [reply frontend/src/Calendar.js]: Got it. I binded the functions into the components constructor [here](https://github.com/googleinterns/step78-2020/pull/43/commits/d82d8120e0cfa8debdafaee44bbc573ea79d61ca) and this should save the overhead you mentioned.

**Review by FabriceCastel — APPROVED** [review]
LGTM - please run the linter again to fix any pending style issues from your last commits before merging.

**Review by kristen-law — APPROVED** [review]

**Pull request merged**
